### PR TITLE
Purchases

### PR DESCRIPTION
**Pull request by SofiaCantero24**
## What does this do?
Purchases screen

https://github.com/user-attachments/assets/c6cac47b-8c41-4fa5-b01a-cc8b982dd92e

## Notion Ticket
* [Purchases screen](https://www.notion.so/rootstrap/My-purchases-bcc62f1be0e84860a99e0026eb2b53ac?pvs=4)

## Figma view
* [Purchases screen](https://www.figma.com/file/uUWXfDpuvDtnOHdSRjv0t1/Black-Market---iOS?type=design&node-id=942-1777&mode=design&t=2OqGoK3yhEqluy8w-4)

## How did you test this?

### CODE REVIEW

File: src/app/(app)/purchases.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }, [refetch])
+  );
+
+  useEffect(() => {
+    const allLineItems =
+      data?.pages.flatMap((page) =>
+        page.data.flatMap((purchase: Purchase) => purchase.line_items)
+      ) || [];
+    setLineItems(allLineItems);
+  }, [data]);
```
Comment (juanchoperezj) [inline src/app/(app)/purchases.tsx]: This effect can be removed, [as reference](https://react.dev/learn/you-might-not-need-an-effect).

I identified two alternatives to get rid of it:
* [select](https://tanstack.com/query/latest/docs/framework/react/reference/useQuery#usequery) prop in the get-purchases infinite query. There, you can transform your data without worrying if it exists; the react query will validate it for you.
* Transforming the data during render, without saving it into state.

File: src/app/(app)/purchases.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import { API_CONSTS } from '@/api/consts';
+import { usePruchases } from '@/api/purchase/get-purchases';
+import type { Purchase } from '@/api/purchase/types';
+import { HeaderLogo } from '@/components/header-logo';
+import { PurchaseCard } from '@/components/purchases/purchase-card';
+import { SafeAreaView, Text, View } from '@/ui';
+
+export default function Purchases() {
+  const { data, hasNextPage, fetchNextPage, isFetchingNextPage, refetch } =
+    usePruchases({
```
Comment (Stukz) [inline src/app/(app)/purchases.tsx]: you have a typo here `usePurchases`

File: src/components/purchases/purchase-card.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+              className='className="h-full w-2/4'
+            >
+              <Text className="text-md font-semibold">{name}</Text>
+            </Link>
+            <Text className="font-semibold">Price: {price}</Text>
+          </View>
+          <StateLabel state={state} />
+          <View className="mt-4 flex-row items-center justify-between">
+            <Text className="text-md font-semibold">{quantity} item</Text>
+            <Text> Bought on</Text>
```
Comment (Stukz) [inline src/components/purchases/purchase-card.tsx]: do you want this extra space ?
```suggestion
            <Text>Bought on</Text>
```

**Review by juanchoperezj — APPROVED** [review]

File: src/components/purchases/purchase-card.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+              className='className="h-full w-2/4'
+            >
+              <Text className="text-md font-semibold">{name}</Text>
+            </Link>
+            <Text className="font-semibold">Price: {price}</Text>
+          </View>
+          <StateLabel state={state} />
+          <View className="mt-4 flex-row items-center justify-between">
+            <Text className="text-md font-semibold">{quantity} item</Text>
+            <Text>Bought on</Text>
```
Comment (fernandatoledo) [inline src/components/purchases/purchase-card.tsx:69]: Bought on? is it supposed to be followed by a date of purchase?

**Pull request merged**